lib/ddtrace/contrib/concurrent_ruby/context_composite_executor_service.rb

```diff
@@ -14,15 +14,24 @@ def initialize(composited_executor)
           @composited_executor = composited_executor
         end
 
-        # post method runs the task within composited executor - in a different thread
-        def post(*args, &task)
+        # The post method runs the task within composited executor - in a
+        # different thread. The original arguments are captured to be
+        # propagated to the composited executor post method
+        def post(*args, &block)
           parent_context = datadog_configuration.tracer.provider.context
 
-          @composited_executor.post(*args) do
+          executor = @composited_executor.is_a?(Symbol) ? Concurrent.executor(@composited_executor) : @composited_executor
```

Comment:
The default executor pools are global but in `Concurrent::Promises` a custom executor instance can be passed as argument in some public API methods.

Reply:
Interesting. If that's the case, how would it affect this line of code (vs if it was executed at initialization time)? e.g. would `@composited_executor` or `Concurrent.executor(@composited_executor)` change based on the promise configuration?

I wouldn't quite expect them too (??).

Reply:
Ah, I believe you are correct, my mind fumbled and for some reason I thought you were referring to another initializer.

Reply:
Hmm in fact I now recall it's the `Concurrent.executor` behaviour I'm more concerned with. Doing the call here is consistent with the moment where the same method is called in `evaluate_to` (it's just a little bit earlier in the call stack), whereas if called in `initialize` the timeline is sensibly different, including being called in another thread entirely. With the current implementation of `concurrent-ruby` it does not look like it could be a problem, but I feel it would be introducing some brittleness.

Reply:
I find that usually things break in the ways we didn't think of (lolol) but it's harmless so I'm happy to keep it -- I just asked to make sure that I wasn't missing anything :)